setup.py

```diff
@@ -1,23 +1,126 @@
+from __future__ import print_function
 
 from setuptools import setup, find_packages
 
+try:
+    from py2app.build_app import py2app
+    py2app_available = True
+except ImportError:
+    py2app_available = False
+
+
+_NAME = "mimic"
+_VERSION = "1.3.0"
+
+
+def setup_options(name, version):
+    """
+    If `py2app` is present in path, then enable to option to build the app.
+
+    This also disables the options needed for normal `sdist` installs.
+
+    :returns: a dictionary of setup options.
+    """
+    info = dict(
+        install_requires=[
+            "characteristic==14.2.0",
+            "klein==0.2.1",
+            "twisted==14.0.0",
+            "jsonschema==2.0",
+            "treq==0.2.0",
+            "six==1.6.1"
+        ],
+        package_dir={"mimic": "mimic"},
+        packages=find_packages(exclude=[]) + ["twisted.plugins"],
+    )
+    if not py2app_available:
+        return info
+
+    # py2app available, proceed.
+    script="bundle/start-app.py"
+    test_script="bundle/run-tests.py"
+    plist = dict(
+        CFBundleName                = _NAME,
+        CFBundleShortVersionString  = " ".join([_NAME, _VERSION]),
+        CFBundleGetInfoString       = _NAME,
+        CFBundleExecutable          = _NAME,
+        CFBundleIdentifier          = "com.%s.%s" % (_NAME, _VERSION),
+        LSUIElement                 = "1",
+        LSMultipleInstancesProhibited = "1",
+    )
+    app_data = dict(
+        script=script,
+        plist=plist,
+        extra_scripts=[test_script]
+    )
+
+    class BuildWithCache(py2app, object):
+        """
+        Before building the application rebuild the `dropin.cache` files.
+        """
+
+        def collect_recipedict(self):
+            """
+            Implement a special Twisted plugins recipe so that dropin.cache
+            files are generated and included in site-packages.zip.
+            """
+            result = super(BuildWithCache, self).collect_recipedict()
+            def check(cmd, mg):
+                from twisted.plugin import getPlugins, IPlugin
+                from twisted import plugins as twisted_plugins
+                from mimic import plugins as mimic_plugins
+
+                for plugin_package in [twisted_plugins, mimic_plugins]:
+                    import time
+                    list(getPlugins(IPlugin, package=plugin_package))
+
+                import os
+                def plugpath(what):
+                    path_in_zip = what + "/plugins"
+                    path_on_fs = (
+                        os.path.abspath(
+                            os.path.join(
+                                os.path.dirname(
+                                    __import__(what + ".plugins",
+                                               fromlist=["nonempty"])
+                                    .__file__),
+                                "dropin.cache")
+                        ))
+                    os.utime(path_on_fs, (time.time() + 86400,) * 2)
```

Comment:
I think I glossed over this piece of code instead of making a comment - sorry about that. I do agree that a comment describing its purpose would be helpful.